### PR TITLE
Task cards

### PR DESCRIPTION
**Pull request by AkshitaB**
Motivation: A model's abilities and limitations come from the underlying task and the dataset it is trained on, in addition to the model itself. Therefore, it is helpful to separate out the information in model cards into 3 different concepts:

Task Card: Issues with defining the task itself. Eg. Span selection as a substitute for reading comprehension.
Data Card/Sheet: Biases in the data, etc. Eg. imbalanced demographics.
Model Card (Model specific): Architecture, algorithmic choices; loss function and metric choices. Eg. differences in picking L2 (average) error and L1 (median) error.

This PR adds a `TaskCard` class and task cards for some tasks in our demo. An additional advantage of this abstraction is that it can be directly used in the demo.

### CODE REVIEW

File: allennlp_models/common/task_card.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    """
+
+    id: str
+    name: Optional[str] = None
+    description: Optional[str] = None
+    expected_inputs: Optional[str] = None
+    expected_outputs: Optional[str] = None
+    scope_and_limitations: Optional[str] = None
+    examples: Optional[Union[List[Dict[str, str]], Dict[str, List[Dict[str, str]]]]] = None
+    # TODO: Add this to model cards.
```
Comment (epwalsh) [inline allennlp_models/common/task_card.py:62]: So will `ModeCard` have a field for `task`?

Comment (AkshitaB) [reply allennlp_models/common/task_card.py:62]: Yes, a reference to `task_id`.

Comment (epwalsh) [reply allennlp_models/common/task_card.py:62]: Can that get added in this PR?

File: tests/pretrained_test.py [hunk 2]
```diff
@@ -407,3 +408,7 @@ def test_transformer_qa(self):
     def test_pretrained_models(self, model_id, model_card):
         # Each model in pretrained_models should have an archive and registered_predictor_name.
         assert model_card.archive_file is not None
+
+    @pytest.mark.parametrize("task_id, task_card", get_tasks().items())
```
Comment (epwalsh) [inline tests/pretrained_test.py:412]: I think we should also have a test that ensures the `task_id` for each model card is valid.

**Review by epwalsh — APPROVED** [review]
